### PR TITLE
#170947422 configure ESlint and babel

### PR DESCRIPTION
**Pull request by AbdoulNuru**
#### What does this PR do?
Have eslint and babel configured.

#### Description of Task to be completed?
- Configure eslint to lint our codes using airbnb as the style guide and all the predefined rules
- Configure babel to convert our ES6 code to the compatible version which is ES5

#### How should this be manually tested?
- Use `git clone <repo link>` to clone the repo to your local machine
- Use `npm i` to install all the project dependencies
- Run `npm run lint` to have eslint report problems, warnings and errors in your code
- Run `npm run lintFix` to have eslint fix all auto fixable problems.

#### What are the relevant pivotal tracker stories?
[#170947422](https://www.pivotaltracker.com/story/show/170947422)
#### Screenshots (if appropriate)
Screen Shot 2020-02-04 at 14 11 04
Screen Shot 2020-02-04 at 14 11 28

#### Questions:

### CODE REVIEW

**Review by RusimbiPatrick — COMMENTED** [review]
Check out the ESlint new line problem

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    "no-param-reassign": 0,
+    "comma-dangle": 0,
+    "curly": ["error", "multi-line"],
+    "import/no-unresolved": [2, { "commonjs": true }],
+    "no-shadow": ["error", { "allow": ["req", "res", "err"] }],
+    "valid-jsdoc": ["error", {
+      "requireReturn": true,
+      "requireReturnType": true,
+      "requireParamDescription": false,
+      "requireReturnDescription": true
```
Comment (izzett222) [inline .eslintrc.json]: hey @AbdoulNuru , the eslint rules above have different indentation

**Review by izzett222 — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 1 above)
Comment (AbdoulNuru) [reply .eslintrc.json]: @izzett222 Because some of them are nested inside the other.

**Review by hakoemmy — APPROVED** [review]

File: .eslintrc.json (hunk 1 above)
Comment (izzett222) [reply .eslintrc.json]: that's true, but apart from the first rules, the others don't have any indentation at all.

**Comment by AbdoulNuru** [comment]
> Check out the ESlint new line problem

@RusimbiP It was a mistake but I fixed it you can check now, thank you for noticing

**Comment by blaise82** [comment]
just a question do we need to add screenshots of our code too (in the PR description)?

for I don't think that code Screenshots are appropriate

**Comment by RusimbiPatrick** [comment]
> just a question do we need to add screenshots of our code too (in the PR description)?

I think it would be helpful.

File: .eslintrc.json (hunk 1 above)
Comment (AbdoulNuru) [reply .eslintrc.json]: @izzett222 I have realized that some rules were not indented correctly and I fixed it, Thanks.

**Review by izzett222 — APPROVED** [review]

**Review by RusimbiPatrick — APPROVED** [review]

**Comment by AbdoulNuru** [comment]
> just a question do we need to add screenshots of our code too (in the PR description)?
> 
> for I don't think that code Screenshots are appropriate

@blaise82 I added them because I think they are helpful, but am not sure if it's appropriate or not, maybe our TTL will help us with this one.

**Comment by hakoemmy** [comment]
> just a question do we need to add screenshots of our code too (in the PR description)?
> 
I think adding screenshots of code is a kind of messy when you have written lots of code but there is no problem with chores.

**Comment by blaise82** [comment]
> > just a question do we need to add screenshots of our code too (in the PR description)?
> 
> I think adding screenshots of code is a kind of messy when you have written lots of code but there is no problem with chores.

I don't think it's appropriate because if I wanted to know what he changed i could just check 
and in the screenshots, it's harder to know what's new and what's old

**Comment by hakoemmy** [comment]
> > > just a question do we need to add screenshots of our code too (in the PR description)?
> > 
> > 
> > I think adding screenshots of code is a kind of messy when you have written lots of code but there is no problem with chores.
> 
> I don't think it's appropriate because if I wanted to know what he changed i could just check
> and in the screenshots, it's harder to know what's new and what's old

>>
> Sure. When you are working on a feature you better attach postman screenshots

**Comment by blaise82** [comment]
maybe one/two important things he might have forgotten is adding build and eslint fix commands to package.json

**Comment by AbdoulNuru** [comment]
@blaise82 am not sure about that

**Comment by blaise82** [comment]
> @blaise82 am not sure about that

morning
about?

**Review by blaise82 — APPROVED** [review]

**Review by kcharles52 — CHANGES_REQUESTED** [review]
Please create a script in the package.json file with a name and can just be run with `npm run <script-name>`
Example
under scripts on package.json file add something like this
`"lint":"command to run linting with eslint"`
such that we only have to type `npm lint or npm run lint`

**Review by kcharles52 — APPROVED** [review]
Welldone

**Pull request merged**